app/src/main/java/com/abhijitvalluri/android/fitnotifications/services/NLService.java

```diff
@@ -443,16 +375,122 @@ private boolean appNotificationScheduleActive(String appPackageName) {
 
         int startTime = appSelection.getStartTime();
         int stopTime = appSelection.getStopTime();
-        Date currDate = new Date();
 
         // Get current time
         Calendar cal = Calendar.getInstance();
-        cal.setTime(currDate);
         int hour = cal.get(Calendar.HOUR_OF_DAY);
         int minute = cal.get(Calendar.MINUTE);
-        int currTime = hour*60 + minute;
+        int currTime = hour * 60 + minute;
+
+        return ((currTime >= startTime) && (currTime <= stopTime));
+    }
+
+    private boolean isScreenOn() {
+        PowerManager pm = (PowerManager) getSystemService(Context.POWER_SERVICE);
 
-        return ((currTime >= startTime) && (currTime < stopTime));
+        if (Build.VERSION.SDK_INT > Build.VERSION_CODES.KITKAT) {
+            // API >= 20
+            return pm.isInteractive();
+        }
+
+        // API <= 19, use deprecated
+        //noinspection deprecation
+        return pm.isScreenOn();
+    }
+
+    /**
+     * Checks if any of the <code>CharSequence</code> items contains the provided <code>filter</code>
+     * text.
+     */
+    private static boolean anyMatchesFilter(String filter, CharSequence ... items) {
+        if (filter != null && !filter.isEmpty()) {
+            String[] parts = filter.split("\\s*,\\s*");
+
+            for (CharSequence item : items) {
+                if (item != null) {
+                    String tmp = item.toString();
+                    for (String filterText : parts) {
+                        if (filterText.length() > 0 && tmp.contains(filterText)) {
+                            return true;
+                        }
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    private String transliterate(CharSequence text) {
+        try {
+            return text == null ? null : transliterator.transform(text.toString());
+        } catch (Exception e) {
+            return text.toString();
+        }
+    }
+
+    private static boolean isNotEmpty(CharSequence text) {
+        return text != null && text.length() > 0;
+    }
+
+    /**
+     * Checks if the <code>big</code> text has the <code>small</code> text as the prefix.
+     * Returns <code>false</code> if <code>small</code> text is empty.
+     */
+    private static boolean startsWith(CharSequence big, CharSequence small) {
+        return isNotEmpty(big) && isNotEmpty(small)
+                && big.subSequence(0, small.length()).toString().contentEquals(small);
+    }
+
+    private static boolean isBlank(CharSequence text) {
+        if (text != null && text.length() > 0) {
+            for (int i = 0; i < text.length(); i++) {
+                // FIXME: isWhitespace() does not recognize some characters (e.g. non-breaking space)
+                if (!Character.isWhitespace(text.charAt(i))) {
+                    return false;
+                }
+            }
+        }
+        return true;
```

Comment:
On one hand you added an extra block to de-allocate `AppSelection` sooner. On the other hand you suggest to create two `String` instances only to check that there is some text.

Should I remove this method and use `StringUtils` from `commons-lang` instead?